[Bug]:  Flink Logical Type ROW is not supported for Array<Tuple> clickhouse type

### What happened?

Clickhouse connector doesn't support Flink Logical Type ROW when I'm trying to ingest a nested Row into 
Array(Tuple)

```
Caused by: java.lang.UnsupportedOperationException: Unsupported type:ROW<`a` INT, `b` INT, `c` INT>
	at org.apache.flink.connector.clickhouse.internal.converter.ClickHouseConverterUtils.toExternal(ClickHouseConverterUtils.java:118)
	at org.apache.flink.connector.clickhouse.internal.converter.ClickHouseConverterUtils.toExternal(ClickHouseConverterUtils.java:96)
```

### Affects Versions

master

### What are you seeing the problem on?

_No response_

### How to reproduce

create a clickhouse table with flatten_nested=0 and Array(Tuple) column
```
set flatten_nested=0;
create table test_tuple3(arr Array(Tuple(a Int32, b Int32, c Int32)), z Int32) Engine=MergeTree() Order By tuple();
```

Flink SQL

```
 CREATE TABLE test_tuple (
    arr Array<Row<a int,b int,c int>>,
    z int
  ) WITH (
    'connector' = 'clickhouse',
    'url' = '......',
    'database-name' = 'default',
    'table-name' = 'test_tuple3',
  .....
  );
  insert into test_tuple values(Array[(1,2,3)],600);
```

Got below error:
```
java.lang.UnsupportedOperationException: Unsupported type:ROW<`a` INT, `b` INT, `c` INT>
	at org.apache.flink.connector.clickhouse.internal.converter.ClickHouseConverterUtils.toExternal(ClickHouseConverterUtils.java:118)
	at org.apache.flink.connector.clickhouse.internal.converter.ClickHouseConverterUtils.toExternal(ClickHouseConverterUtils.java:96)
	at org.apache.flink.connector.clickhouse.internal.converter.ClickHouseRowConverter.lambda$createToExternalConverter$e45b375b$1(ClickHouseRowConverter.java:228)
	at org.apache.flink.connector.clickhouse.internal.converter.ClickHouseRowConverter.toExternal(ClickHouseRowConverter.java:95)
	at org.apache.flink.connector.clickhouse.internal.executor.ClickHouseBatchExecutor.addToBatch(ClickHouseBatchExecutor.java:80)
	at org.apache.flink.connector.clickhouse.internal.ClickHouseBatchOutputFormat.writeRecord(ClickHouseBatchOutputFormat.java:97)
	at org.apache.flink.connector.clickhouse.internal.ClickHouseBatchOutputFormat.writeRecord(ClickHouseBatchOutputFormat.java:33)
	at org.apache.flink.streaming.api.functions.sink.OutputFormatSinkFunction.invoke(OutputFormatSinkFunction.java:104)
	at org.apache.flink.streaming.api.functions.sink.SinkFunction.invoke(SinkFunction.java:50)
	at org.apache.flink.table.runtime.operators.sink.SinkOperator.processElement(SinkOperator.java:65)
	at org.apache.flink.streaming.runtime.tasks.ChainingOutput.pushToOperator(ChainingOutput.java:94)
	at org.apache.flink.streaming.runtime.tasks.ChainingOutput.collect(ChainingOutput.java:75)
	at org.apache.flink.streaming.runtime.tasks.ChainingOutput.collect(ChainingOutput.java:39)
	at BatchExecCalc$6.processElement(Unknown Source)
```


### Relevant log output

_No response_

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/flink-connector-clickhouse/src/main/java/org/apache/flink/connector/clickhouse/internal/converter/ClickHouseConverterUtils.java
+++ b/flink-connector-clickhouse/src/main/java/org/apache/flink/connector/clickhouse/internal/converter/ClickHouseConverterUtils.java
@@ -21,7 +21,9 @@ import org.apache.flink.table.data.ArrayData;
 import org.apache.flink.table.data.DecimalData;
 import org.apache.flink.table.data.GenericArrayData;
 import org.apache.flink.table.data.GenericMapData;
+import org.apache.flink.table.data.GenericRowData;
 import org.apache.flink.table.data.MapData;
+import org.apache.flink.table.data.RowData;
 import org.apache.flink.table.data.StringData;
 import org.apache.flink.table.data.TimestampData;
 import org.apache.flink.table.types.logical.ArrayType;
@@ -33,6 +35,8 @@ import com.clickhouse.data.value.UnsignedByte;
 import com.clickhouse.data.value.UnsignedInteger;
 import com.clickhouse.data.value.UnsignedLong;
 import com.clickhouse.data.value.UnsignedShort;
+
+import org.apache.flink.table.types.logical.RowType;
 
 import java.math.BigDecimal;
 import java.math.BigInteger;
@@ -46,7 +50,9 @@ import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
 import java.time.OffsetDateTime;
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.UUID;
 
@@ -119,9 +125,19 @@ public class ClickHouseConverterUtils {
                             toExternal(valueGetter.getElementOrNull(valueArrayData, i), valueType));
                 }
                 return objectMap;
-            case MULTISET:
             case ROW:
+                List<Object> result = new ArrayList<>();
+                for (int i = 0; i < ((RowData) value).getArity(); i++) {
+                    result.add(
+                            toExternal(
+                                    RowData.createFieldGetter(
+                                                    ((RowType) type).getTypeAt(i), i)
+                                            .getFieldOrNull((RowData) value),
+                                    ((RowType) type).getTypeAt(i)));
+                }
+                return result;
             case RAW:
+            case MULTISET:
             default:
                 throw new UnsupportedOperationException("Unsupported type:" + type);
         }
@@ -209,6 +225,12 @@ public class ClickHouseConverterUtils {
                 }
                 return new GenericMapData(internalMap);
             case ROW:
+                List<Object> row = (List<Object>) value;
+                GenericRowData rowData = new GenericRowData(row.size());
+                for (int i = 0; i < row.size(); i++) {
+                    rowData.setField(i, toInternal(row.get(i), type.getChildren().get(i)));
+                }
+                return rowData;
             case MULTISET:
             case RAW:
             default:

--- a/flink-connector-clickhouse/src/main/java/org/apache/flink/connector/clickhouse/internal/converter/ClickHouseConverterUtils.java
+++ b/flink-connector-clickhouse/src/main/java/org/apache/flink/connector/clickhouse/internal/converter/ClickHouseConverterUtils.java
@@ -30,13 +30,12 @@ import org.apache.flink.table.types.logical.ArrayType;
 import org.apache.flink.table.types.logical.DecimalType;
 import org.apache.flink.table.types.logical.LogicalType;
 import org.apache.flink.table.types.logical.MapType;
+import org.apache.flink.table.types.logical.RowType;
 
 import com.clickhouse.data.value.UnsignedByte;
 import com.clickhouse.data.value.UnsignedInteger;
 import com.clickhouse.data.value.UnsignedLong;
 import com.clickhouse.data.value.UnsignedShort;
-
-import org.apache.flink.table.types.logical.RowType;
 
 import java.math.BigDecimal;
 import java.math.BigInteger;
@@ -130,8 +129,7 @@ public class ClickHouseConverterUtils {
                 for (int i = 0; i < ((RowData) value).getArity(); i++) {
                     result.add(
                             toExternal(
-                                    RowData.createFieldGetter(
-                                                    ((RowType) type).getTypeAt(i), i)
+                                    RowData.createFieldGetter(((RowType) type).getTypeAt(i), i)
                                             .getFieldOrNull((RowData) value),
                                     ((RowType) type).getTypeAt(i)));
                 }

--- a/flink-connector-clickhouse/src/main/java/org/apache/flink/connector/clickhouse/util/DataTypeUtil.java
+++ b/flink-connector-clickhouse/src/main/java/org/apache/flink/connector/clickhouse/util/DataTypeUtil.java
@@ -17,6 +17,7 @@
 
 package org.apache.flink.connector.clickhouse.util;
 
+import org.apache.flink.api.java.tuple.Tuple2;
 import org.apache.flink.table.api.DataTypes;
 import org.apache.flink.table.catalog.exceptions.CatalogException;
 import org.apache.flink.table.types.DataType;
@@ -25,6 +26,7 @@ import com.clickhouse.data.ClickHouseColumn;
 
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 import static org.apache.flink.table.types.logical.DecimalType.MAX_PRECISION;
 
@@ -110,6 +112,12 @@ public class DataTypeUtil {
                         toFlinkType(clickHouseColumnInfo.getKeyInfo()),
                         toFlinkType(clickHouseColumnInfo.getValueInfo()));
             case Tuple:
+                return DataTypes.ROW(
+                        clickHouseColumnInfo.getNestedColumns().stream()
+                                .map((col) -> new Tuple2<>(col, toFlinkType(col)))
+                                .map(tuple -> DataTypes.FIELD(tuple.f0.getColumnName(), tuple.f1))
+                                .collect(Collectors.toList()));
+
             case Nested:
             case AggregateFunction:
             default:

